nilearn/decoding/space_net.py

```diff
@@ -240,18 +239,15 @@ def __call__(self, variables):
             len(self.test_scores) > 4
             and np.mean(np.diff(self.test_scores[-5:][::-1])) >= self.tol
         ):
+            message = "."
             if self.verbose:
-                if self.verbose > 1:
-                    print(
-                        "Early stopping. "
-                        f"Test score: {score:.8f} {40 * '-'}"
-                    )
-                else:
-                    sys.stderr.write(".")
+                message = (
+                    f"Early stopping. \n" f"Test score: {score:.8f} {40 * '-'}"
+                )
+            logger.log(message)
             return True
 
-        if self.verbose > 1:
```

Comment:
TODO: some part of the code have a different default value for verbosity, make sure the printing behavior is unchanged.

Reply:
TODO: need to add msg_lvl